### DISCUSSION TITLE
Enable '@' to be used in a library name

### DISCUSSION THREAD

**Discussion by peinearydevelopment** [angular-eslint/angular-eslint#942] [Q&A] [answered]
In trying to utilize angular-eslint for my company, and following the pattern I've seen in many angular related libraries(including this one), I tried to give my library the name `@company-name`. When I do that, I'm getting a linting error that I assume is coming from [this schematic](https://github.com/angular-eslint/angular-eslint/blob/4c11177e66cb54ba8d548ad7068990679c0c7e93/packages/schematics/src/library/schema.json#L13). It would appear as though this was a deliberate decision. Would it be possible to explain why this practice is discouraged? Alternatively, could that part of the regex be dropped enabling this type of a naming of libraries without linting issues?

**Comment by peinearydevelopment** [comment]
After reviewing the full RegEx used there, it appears as though `@company-name` isn't recommended, but `@company-name/foo` is allowed. So is the `@` symbol meant to represent an organization or grouping of libraries, but the expectation is then that there needs to be another modifier after the `/` that will specify this libraries specific intent within that grouping?

**Comment by JamesHenry** [comment, marked as answer]
This matches the pattern used by package managers - @company-name represents an organization and @company-name/foo is a package under that organization. It is therefore not possible to have a package called @company-name on npm, for example, and I think keeping alignment with that mental model is useful for folks far more often than not
